src/path.cr

```diff
@@ -1086,6 +1110,27 @@ struct Path
     @kind <=> other.@kind
   end
 
+  # Returns `true` if this path is considered equivalent to *other*.
+  #
+  # The comparison is performed strictly lexically: `foo` and `./foo` are *not*
+  # treated as equal. Nor are paths of different `kind`.
+  # To compare paths semantically, they need to be normalized and converted to
+  # the same kind.
+  #
+  # ```
+  # Path["foo"] == Path["foo"]               # => true
+  # Path["foo"] == Path["./foo"]             # => false
+  # Path["foo"] == Path["foo/"]              # => false
+  # Path.posix("foo") == Path.windows("foo") # => false
```

Comment:
Hmm this sample code is a bit misleading because it says `==` but then only after a big chunk of whitespace it says "actually that's false".

On the other hand, if you change this to `!=`, then adding `# => true` at the end is also misleading.

Maybe just have to drop the comments at the end..

Reply:
I see what you mean, but IMO this is the least surprising variant. Besides, we use this pattern everywhere for documentation. I don't think we should change anything. Using `!=` doesn't help, because the visual difference is even less than with `==` and `false`.

Reply:
OK that's fine.
Could still try to minimize the whitespace.
```suggestion
  # Path["foo"] == Path["foo"]   # => true
  # Path["foo"] == Path["./foo"] # => false
  # Path["foo"] == Path["foo/"]  # => false
  #
  # Path.posix("foo") == Path.windows("foo") # => false
```

Reply:
I don't know. Is the whitespace really such a visual barrier in the rendered API docs? IMO adding different indents might add confusion, as well as the random empty line, just to force the formatted.
All these expressions form a set. The last one is just visually different because it uses explicitly named constructor methods. But semantically, there should be no distance to the previous ones.

We could consider using explicit `.new` methods instead of the `.[]` short form. That would reduce the difference in lenght between the lines.